src/main/java/org/verdictdb/core/querying/simplifier/ConsolidatedExecutionNode.java

```diff
@@ -229,6 +241,28 @@ public SqlConvertible createQuery(List<ExecutionInfoToken> tokens) throws Verdic
   public ExecutionInfoToken createToken(DbmsQueryResult result) {
     // pass the information from the internal parent node
     ExecutionInfoToken token = parentNode.createToken(result);
+    // Addition check that the query is a query contains Asterisk column that without asyncAggExecutionNode.
+    // For instance, query like 'select * from lineitem'. In that case, all the values of isAggregate field
+    // are false.
+    if (token.containsKey("queryResult")) {
+      DbmsQueryResult queryResult = (DbmsQueryResult) token.getValue("queryResult");
+      if (queryResult.getColumnCount() != queryResult.getMetaData().isAggregate.size()) {
+        List<Boolean> isAggregate = new ArrayList<>();
+        for (int i = 0; i < queryResult.getColumnCount(); i++) {
+          isAggregate.add(false);
+        }
+        for (int i = 0; i < queryResult.getMetaData().isAggregate.size(); i++) {
+          if (!selectQueryColumnAlias.get(i).equals(asteriskAlias)) {
+            int idx = 0;
+            while (!selectQueryColumnAlias.get(i).equals(queryResult.getColumnName(idx))) {
```

Comment:
Can you add a comment about this line? What is this condition (inside the parenthesis) effectively testing?